cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -41,42 +68,97 @@
 @Singleton
 public class StreamSizeScheduler implements Scheduler {
   private static final Logger LOG = LoggerFactory.getLogger(StreamSizeScheduler.class);
+  private static final int STREAM_POLLING_THREAD_POOL_SIZE = 10;
 
+  private final long pollingDelay;
   private final NotificationService notificationService;
   private final StreamAdmin streamAdmin;
   private final StoreFactory storeFactory;
   private final ProgramRuntimeService programRuntimeService;
   private final PreferencesStore preferencesStore;
+  private final ConcurrentMap<Id.Stream, StreamSubscriber> streamSubscribers;
 
+  // Key is scheduleId
+  private final ConcurrentMap<String, StreamSubscriber> scheduleSubscribers;
+
+  private Store store;
+  private Executor notificationExecutor;
+  private ScheduledExecutorService streamPollingExecutor;
 
   @Inject
   public StreamSizeScheduler(CConfiguration cConf, NotificationService notificationService, StreamAdmin streamAdmin,
                              StoreFactory storeFactory, ProgramRuntimeService programRuntimeService,
                              PreferencesStore preferencesStore) {
+    this.pollingDelay = cConf.getLong(Constants.Notification.Stream.STREAM_SIZE_SCHEDULE_POLLING_DELAY);
     this.notificationService = notificationService;
     this.streamAdmin = streamAdmin;
     this.storeFactory = storeFactory;
     this.programRuntimeService = programRuntimeService;
     this.preferencesStore = preferencesStore;
+    this.streamSubscribers = Maps.newConcurrentMap();
+    this.scheduleSubscribers = Maps.newConcurrentMap();
+    this.store = null;
   }
 
   public void start() {
-
+    notificationExecutor = Executors.newCachedThreadPool(Threads.createDaemonThreadFactory("stream-size-scheduler-%d"));
+    streamPollingExecutor = Executors.newScheduledThreadPool(STREAM_POLLING_THREAD_POOL_SIZE,
+                                                             Threads.createDaemonThreadFactory("stream-polling-%d"));
   }
 
   public void stop() {
-
+    streamPollingExecutor.shutdownNow();
+    for (StreamSubscriber subscriber : streamSubscribers.values()) {
+      subscriber.cancel();
+    }
   }
 
   @Override
   public void schedule(Id.Program program, SchedulableProgramType programType, Schedule schedule) {
     Preconditions.checkArgument(schedule instanceof StreamSizeSchedule,
                                 "Schedule should be of type StreamSizeSchedule");
+    StreamSizeSchedule streamSizeSchedule = (StreamSizeSchedule) schedule;
+    schedule(program, programType, streamSizeSchedule, true, -1, -1, true);
+  }
+
+  private void schedule(Id.Program program, SchedulableProgramType programType, StreamSizeSchedule streamSizeSchedule,
```

Comment:
I will add. baseRunSize is the size from which a schedule starts counting an increment - which is likely to be the size that we had the last time the schedule was triggered. baseRunTs is the time at which that baseRunSize was obtained.